File: Python/11-invert-values.py
# Invert values
# Difficulty: 8 kyu

def invert(lst):
    return [-i for i in lst]


# Test

import codewars_test as test
from solution import invert

@test.describe("Invert values")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(invert([1,2,3,4,5]),[-1,-2,-3,-4,-5])
        test.assert_equals(invert([1,-2,3,-4,5]), [-1,2,-3,4,-5])
        test.assert_equals(invert([]), [])

File: R/8kyu/11-invert-values.r
# Invert values
# Difficulty: 8 kyu

# Soltution 1
invert <- function(vec){
  vec * -1
}

# Solution 2 (more concise)
invert <- function(vec){
  -vec
}


# Test

test_that('Basic tests', {
  expect_equal(invert(c(1,2,3,4,5)),c(-1,-2,-3,-4,-5))
  expect_equal(invert(c(1,-2,3,-4,5)), c(-1,2,-3,4,-5))
  expect_equal(invert(numeric()), numeric())
  expect_equal(invert(c(0)), c(0))
})